src/jdk.jlink/share/classes/jdk/tools/jlink/internal/TaskHelper.java

```diff
@@ -535,18 +535,22 @@ public List<String> handleOptions(T task, String[] args) throws BadArgs {
                     }
                     Option<?> opt = pluginOption == null ? option : pluginOption;
                     String param = null;
+                    boolean potentiallyGnuOption = false;
                     if (opt.hasArg) {
                         if (name.startsWith("--") && name.indexOf('=') > 0) {
                             param = name.substring(name.indexOf('=') + 1,
                                     name.length());
                         } else if (i + 1 < args.length) {
+                            potentiallyGnuOption = true;
                             param = args[++i];
                         }
-                        if (param == null || param.isEmpty()
-                                || (param.length() >= 2 && param.charAt(0) == '-'
-                                && param.charAt(1) == '-')) {
-                            throw new BadArgs("err.missing.arg", name).
-                                    showUsage(true);
+                        if (param == null || param.isEmpty()) {
+                            throw new BadArgs("err.missing.arg", name).showUsage(true);
+                        }
+                        if (potentiallyGnuOption && param.length() >= 2 &&
+                            param.charAt(0) == '-' && param.charAt(1) == '-' &&
+                            !param.contains(" ")) {
```

Comment:
--style argument still could be a real jlink argument, better check would be to see if that's a valid jlink option rather than just "--". 
The other alternative would be like what proposed by @YaSuenag, simply take the next token as value.

Reply:
I would expect the next token would be parsed as the value.  I suggest to explore that and see how complex it would be.

It seems awkward  if the following are accepted:
```
--add-options "--add-modules jdk.incubator.concurrent"
--add-options=--add-modules=jdk.incubator.concurrent
```

but not this:
```
--add-options "--add-modules=jdk.incubator.concurrent"
```